install_script_opw.sh

```diff
@@ -0,0 +1,644 @@
+            exit $apt_exit_code
+        else
+            break
+        fi
+    done
+
+    printf "\033[34m\n* Installing APT package sources for Datadog\n\033[0m\n"
+    $sudo_cmd sh -c "echo 'deb [signed-by=${apt_usr_share_keyring}] https://${apt_url}/ ${apt_repo_version}' > /etc/apt/sources.list.d/datadog.list"
+
+    if [ ! -f $apt_usr_share_keyring ]; then
+        $sudo_cmd touch $apt_usr_share_keyring
+    fi
+    # ensure that the _apt user used on Ubuntu/Debian systems to read GPG keyrings
+    # can read our keyring
+    $sudo_cmd chmod a+r $apt_usr_share_keyring
+
+    for key in "${APT_GPG_KEYS[@]}"; do
+        $sudo_cmd curl --retry 5 -o "/tmp/${key}" "https://${keys_url}/${key}"
+        $sudo_cmd cat "/tmp/${key}" | $sudo_cmd gpg --import --batch --no-default-keyring --keyring "$apt_usr_share_keyring"
+    done
+
+    release_version="$(grep VERSION_ID /etc/os-release | cut -d = -f 2 | xargs echo | cut -d "." -f 1)"
+    if { [ "$DISTRIBUTION" == "Debian" ] && [ "$release_version" -lt 9 ]; } || \
+       { [ "$DISTRIBUTION" == "Ubuntu" ] && [ "$release_version" -lt 16 ]; }; then
+        # copy with -a to preserve file permissions
+        $sudo_cmd cp -a $apt_usr_share_keyring $apt_trusted_d_keyring
+    fi
+
+    ERROR_MESSAGE="ERROR
+Failed to update the sources after adding the Datadog repository.
+This may be due to any of the configured APT sources failing -
+see the logs above to determine the cause.
+If the failing repository is Datadog, please contact Datadog support.
+*****
+"
+    $sudo_cmd apt-get update -o Dir::Etc::sourcelist="sources.list.d/datadog.list" -o Dir::Etc::sourceparts="-" -o APT::Get::List-Cleanup="0"
+    ERROR_MESSAGE="ERROR
+Failed to install one or more packages, sometimes it may be
+due to another APT source failing. See the logs above to
+determine the cause.
+If the cause is unclear, please contact Datadog support.
+*****
+"
+
+    if [ -n "$worker_minor_version" ]; then
+        # Example: observability-pipelines-worker=1.2.1-1
+        pkg_pattern="([[:digit:]]:)?$worker_major_version\.${worker_minor_version%.}(\.[[:digit:]]+){0,1}(-[[:digit:]])?"
+        agent_version_custom="$(apt-cache madison observability-pipelines-worker | grep -E "$pkg_pattern" -om1)" || true
+        verify_agent_version "="
+    fi
+
+    declare -a packages
+    packages=("observability-pipelines-worker" "datadog-signing-keys")
+
+    echo -e "  \033[33mInstalling package(s): ${packages[*]}\n\033[0m"
+
+    $sudo_cmd apt-get install -y --force-yes "${packages[@]}"
+
+    ERROR_MESSAGE=""
+else
+    ERROR_MESSAGE="Your OS or distribution are not supported by this install script.
+Please follow the instructions on the Observability Pipelines Worker setup page:
+
+https://docs.datadoghq.com/observability_pipelines/setup/"
+    ERROR_CODE=$UNSUPPORTED_PLATFORM_CODE
+    printf "\033[31m$ERROR_MESSAGE\033[0m\n"
+    report_telemetry
+    exit;
+fi
+
+# Set the configuration
+if [ "$apikey" ]; then
```

Comment:
I agree with the rule there, it gets a little tricky given how OPW has been packaged - when installed via either linux package the `bootstrap.yaml` file here will always be present - we don't have option to do the "copy the example into place if not present" logic which makes the "first time run" more difficult to manage.

I _guess_ we could install `yq` as a dependency, but I think I'd prefer to avoid that to keep the install footprint minimal.

What about including a machine readable comment indicating "this config file was written to by the install script, skip further changes"?

Reply:
🤔 adding a comment the first time doesn't really work either - we _might_ need to adjust the packaging itself to better model this guideline.